src/classes/CountryStateDistrict.cls

```diff
@@ -0,0 +1,115 @@
+            }
+        }
+    }
+    // Calculate Total number of States And Districts using Aggregate Queries
+    List<Sobject> sobjectListForDml=new List<Sobject> ();
+    public void totalNoOfStateAndDistrict(){
+        List<AggregateResult> stateAndDistrictAggregateResult=[SELECT Country_Name__c country_Id, COUNT(Id) Total_No_States,
+                                                               SUM(Total_Number_of_Districts__c)Total_No_Districts FROM State__c GROUP BY Country_Name__c]; 
+        for(AggregateResult result : stateAndDistrictAggregateResult){
+            Country__c obj= new Country__c();
+            obj.Total_Number_of_States__c=(Integer)result.get('Total_No_States');
+            obj.Total_Number_of_Districts__c=(Decimal)result.get('Total_No_Districts');
+            obj.Id=(Id)result.get('country_Id');
+            sobjectListForDml.add(obj);
+        }
+        if(!sobjectListForDml.isEmpty()){
+            update sobjectListForDml;
+        }
+    }
+    // Calculate Total number of Districts using Aggregate Queries
+    public void totalNoOfDistrict(){
+        
+        List<AggregateResult> DistrictAggregateResult=[SELECT State_Name__c state_Id,COUNT(Id)   
+                                                       Total_No_Districts FROM District__c GROUP BY State_Name__c];
+        for(AggregateResult result : DistrictAggregateResult){
+            State__c obj=new State__c();
+            obj.Total_Number_of_Districts__c=(Integer)result.get('Total_No_Districts');
+            obj.Id=(Id)result.get('state_Id');
+            sobjectListForDml.add(obj); 
+        }
+        if(!sobjectListForDml.isEmpty()){
+            update sobjectListForDml;
+        }
+    }
+    //CountryJSON Generator  method 
+    public void  countryJSONGenerator(string countryName){
+        List<Country__c> countryStateDistrictList=[SELECT Id,Name,Country_Code__c,Total_Number_of_States__c, 
+                                                   Total_Number_of_Districts__c,(SELECT Id,Name,State_Code__c,Total_Number_of_Districts__c FROM States__r) FROM Country__c
+                                                   WHERE Name = : countryName];
+        
+        if(!countryStateDistrictList.isEmpty()){
+            Country wrapperCountryObj= new Country();
+            wrapperCountryObj.countryName=countryStateDistrictList[0].Name;
+            wrapperCountryObj.countryCode=countryStateDistrictList[0].Country_Code__c;
+            wrapperCountryObj.Total_Number_of_States=(Integer)countryStateDistrictList[0].Total_Number_of_States__c;
+            wrapperCountryObj.Total_Number_of_Districts=(Integer)countryStateDistrictList[0].Total_Number_of_Districts__c ;
+            wrapperCountryObj.states=new List<States>();
+            if(!countryStateDistrictList[0].States__r.isEmpty()){
+                for(State__c state:countryStateDistrictList[0].States__r){
+                    States wrapperStateObj=new States();
```

Comment:
Do not create a separate Object for the State wrapper. this object should be made from the country wrapper's constructor(Parameteric)